object_storage_api/stores/image.py

```diff
@@ -37,3 +38,22 @@ def upload(self, image_id: str, image_metadata: ImagePostMetadataSchema, upload_
         )
 
         return object_key
+
+    def create_presigned_url(self, object_key: str, file_name: str) -> HttpUrl:
+        """Generate a presigned URL to share an S3 object.
+
+        :param object_key: object key of the image to generate the url for.
+        :return: Image Metadata with a presigned url.
```

Comment:
Missing `param: file_name`. You could even just take in the `ImageOut` instead and let it use the parameters directly. For attachments it takes in the attachment post object and returns the object key as I wanted to leave all logic on what to use for the presigned url inside the function itself.